dartagnan/src/main/java/com/dat3m/dartagnan/asserts/AssertBasic.java

```diff
@@ -34,7 +36,12 @@ public AbstractAssert removeLocAssertions(boolean replaceByTrue) {
 
     @Override
     public BooleanFormula encode(SolverContext ctx) {
-        return op.encode(e1.getLastValueExpr(ctx), e2.getLastValueExpr(ctx), ctx);
+    	Formula f1 = e1 instanceof LastValueInterface ? ((LastValueInterface)e1).getLastValueExpr(ctx) : null;
+    	Formula f2 = e2 instanceof LastValueInterface ? ((LastValueInterface)e2).getLastValueExpr(ctx) : null;
+    	if(f1 == null || f2 == null) {
+    		throw new RuntimeException("AssertBasic can only relate LastValueInterface objects: " + this);
+    	}
+        return op.encode(f1, f2, ctx);
```

Comment:
I think you could do the check for `LastValueInterface` directly in the constructor.
Also, you can use `Preconditions.checkState` (if you check the state of the current object like in this case) or `Preconditions.checkArgument` if you check the arguments provided (if you move the code to the Constructor).